From 13759b43514b83afe09999865c642e8e134efa7c Mon Sep 17 00:00:00 2001
From: nl6720 <nl6720@gmail.com>
Date: Tue, 25 Oct 2022 18:21:15 +0300
Subject: [PATCH] functions: provide /etc/os-release

Provide /etc/os-release and create a symbolic link from /etc/initrd-release to it.

Fixes https://bugs.archlinux.org/task/67454
---
 functions | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/functions b/functions
index 54c6ee3..068f147 100644
--- a/functions
+++ b/functions
@@ -793,8 +793,19 @@ initialize_buildroot() {
     ln -s /proc/self/mounts "$buildroot/etc/mtab"
     >"$buildroot/etc/fstab"
 
-    # indicate that this is an initramfs
-    >"$buildroot/etc/initrd-release"
+    # add os-release and initrd-release for systemd
+    if [[ -e /etc/os-release ]]; then
+        if [[ -L /etc/os-release ]]; then
+            install -Dm0644 /etc/os-release "${buildroot}$(realpath -- /etc/os-release)"
+            ln -sT  "$(realpath -- /etc/os-release)" "$buildroot/etc/os-release"
+            ln -sT "$(realpath -- /etc/os-release)" "$buildroot/etc/initrd-release"
+        else
+            install -Dm0644 /etc/os-release "$buildroot/etc/os-release"
+            ln -sT os-release "$buildroot/etc/initrd-release"
+        fi
+    else
+        >"$buildroot/etc/initrd-release"
+    fi
 
     # add a blank ld.so.conf to keep ldconfig happy
     >"$buildroot/etc/ld.so.conf"
